### DISCUSSION TITLE
MDF Application & Library Version 2.3

### DISCUSSION THREAD

**Discussion by ihedvall** [ihedvall/mdflib#107] [Announcements]
This release is a minor release. The following major changes have been done.
- Optimized read speed.
- 32-bit build for the .NET library.
- Local timestamp support.

<hr /><em>This discussion was created from the release <a href='https://github.com/ihedvall/mdflib/releases/tag/v2.3.0'>MDF Application & Library Version 2.3</a>.</em>

**Comment by nealwang123** [comment]
can i use the cpp lib with msvc2019 x64?

**Reply by nealwang123** [reply]
The MdfLib was installed by mdflib.exe now.
should i re complie with src code?

**Comment by ihedvall** [comment]
@nealwang123 
I have not personally test to compile with the VS 2019 compiler but it should work well.

The mdflib.exe is just an installation kit mainly intended for the MDF related executables. If you want to include the library into your application, I recommend that you clone the GitHub repository and build it. It's gives you a better maintainability in the long run but this is a minor problem.

The preferred way to build is to use the CMAKE application but there also exist VC Studio project files (*.sln/*.vcxproj). The latter is dependent on VC 2022 Studio but it's not the end of the world to create for the 2019  Studio. The main problem is that the mdflib is dependent on 2 third-party libraries, expat and zlib. These libraries are either built by the VCPKG package manager or manually.

If you not sticking to the VCPKG, you have to do its work manually. There exist a GitHub repository (ihedvall/components) that have most of the batch files needed.
